include/highfive/bits/H5Dataspace_misc.hpp

```diff
@@ -125,100 +125,16 @@ inline std::vector<size_t> DataSpace::getMaxDimensions() const {
     return details::to_vector_size_t(maxdims);
 }
 
-template <typename ScalarValue>
-inline DataSpace DataSpace::From(const ScalarValue& scalar) {
-    (void)scalar;
-    static_assert(
-        (std::is_arithmetic<ScalarValue>::value ||
-         std::is_enum<ScalarValue>::value ||
-         std::is_same<std::string, ScalarValue>::value),
-        "Only the following types are supported by DataSpace::From: \n"
-        "  signed_arithmetic_types = int | long | float | double \n"
-        "  unsigned_arithmetic_types = unsigned signed_arithmetic_types \n"
-        "  string_types = std::string \n"
-        "  all_basic_types = string_types | unsigned_arithmetic_types | "
-        "signed_arithmetic_types \n "
-        "  stl_container_types = std::vector<all_basic_types> "
-        "  boost_container_types = "
-        "boost::numeric::ublas::matrix<all_basic_types> | "
-        "boost::multi_array<all_basic_types> \n"
-        "  all_supported_types = all_basic_types | stl_container_types | "
-        "boost_container_types"
-        "  eigen_matrix_type = Eigen::Matrix<signed_arithmetic_type> | Eigen::VectorXX");
-    return DataSpace(DataSpace::datascape_scalar);
+template <typename T>
+inline DataSpace DataSpace::From(const T& value) {
+    return DataSpace(details::manipulator<T>::get_sizes(value));
```

Comment:
get_sizes() -> get_dimensions?

Reply:
or just `dimensions()`?

Reply:
DataSpace API has `getDimensions()`. no need to be strictly consistent since this is internal API, but still...

Reply:
Resolved in the previous patch.